data-raw/units-data.R

```diff
@@ -31,4 +31,16 @@ halounits <- 'Gg'
 unitstable <- rbind(data.frame(variable=hvars, units=hunits),
                     data.frame(variable=haloemis, units=halounits))
 
-devtools::use_data(unitstable, internal=TRUE, overwrite=TRUE)
+###### Get all available hector variables (as defined by data-raw/vars_all.txt)
+vars_all <- read.table("data-raw/vars_all.txt", header=TRUE, as.is=TRUE)
+
+# Split the variables into two lists: Those who can handle a date parameter
+# and those who cannot
+vars_date <- vars_all[vars_all$date_param == TRUE,][,1]
+vars_nodate <- vars_all[vars_all$date_param == FALSE,][,1]
+
+# Convert the strings representing variable names to capability strings
+vars_date <- lapply(vars_date, do.call, args=list())
```

Comment:
Hmm.  We're taking whatever is in that file and executing it as a function.  That makes me nervous, even if we are theoretically controlling the contents of the file we read the strings from.  Could we explicitly fetch names from the Hector namespace?  That would at least limit the amount of mischief that could be gotten into.  Something like
```R
vfuns <- lapply(vars_date, getFromNamespace, ns='hector')
vars_date <- sapply(bar, do.call, args=list())
```

Reply:
This is a great idea. I didn't know about `getFromNamespace` -- that's very useful! We should do this.

As an aside, if you wanted extra type- and structure-safety, you could write the second line of that as...

```r
vars_date <- vapply(vfuns, do.call, character(1), args=list())
```

...which ensures that every applied call returns a length-1 character vector and throws an informative error if not. That might be particularly useful here in case the wrong thing slips into the variable list file.

Reply:
Addressed in ec76fb94303245b1785bde35fa1c6c15a31cbae1